File: JAVA/Diagonal sum in binary tree.java
class Tree {
    public static ArrayList <Integer> diagonalSum(Node root) 
    {
        ArrayList<Integer> ans  = new ArrayList<>();
        Queue<Node> q = new LinkedList<>();
        q.add(root);
       while(!q.isEmpty())
       {
           int size = q.size();
           int sum = 0;
           for(int i=0;i<size;i++)
           {
               Node curr = q.poll();
               while(curr!=null)
               {
                   if(curr.left!=null)
                     q.add(curr.left);
                   sum += curr.data; 
                   curr = curr.right;
               }
                 
           }
           ans.add(sum);
       }
       return ans;
    }
}

File: C++/Diagonal sum in binary tree.cpp
class Solution {
  public:
    vector<int> diagonalSum(Node* root) {
        // Add your code here
        queue<Node*>q;
        q.push(root);
        vector<int>v;
        while(!q.empty()){
            int n=q.size();class Solution {
  public:
    vector<int> diagonalSum(Node* root) {
        // Add your code here
        queue<Node*>q;
        q.push(root);
        vector<int>v;
        while(!q.empty()){
            int n=q.size();
            int sum=0;
            while(n--){
                Node *j=q.front();
                while(j){
                    q.push(j->left);
                    sum+=j->data;
                    j=j->right;
                }
                q.pop();
            }
            v.push_back(sum);
        }
        v.pop_back();
        return v;
    }
};
            int sum=0;
            while(n--){
                Node *j=q.front();
                while(j){
                    q.push(j->left);
                    sum+=j->data;
                    j=j->right;
                }
                q.pop();
            }
            v.push_back(sum);
        }
        v.pop_back();
        return v;
    }
};
